ipfx/epochs.py

```diff
@@ -108,12 +108,14 @@ def get_stim_epoch(i, test_pulse=True):
     """
 
     di = np.diff(i)
+    di = di[~np.isnan(di)] # != NaN
```

Comment:
This works as a quick fix to unblock you, but long term we should get epochs from the sweep table in the nwb.
Couple of issues:
1. It makes more sense to remove np.nan from the `i` array rather than from the processed `di`
2. If np.nan were to appear in the middle of an array like so:
[0, 0, 1, 1, 0, np.nan, 0, 2, 2, 2, 0, 0, np.nan, np.nan]
then you fix would also drop it. Not sure if that is a feasible case, but if it is, then it is better to just drop the trailing np.nan only and let the code fail down stream to catch this case. Something like this should work: 
```python
import ipfx.time_series_utils as tsu
end_idx = tsu.flatnotnan(i)[-1]
i=i[0:end_idx]


```

Reply:
Thanks for the reply @sgratiy, and I agree with you about issue 1 that it makes more sense to remove np.nan from i rather than di. 

I also see your point for issue 2, although I don't believe there is anyway that we can have np.nan points in the middle of an array with data points before and after; we should only ever have trailing np.nan points, but I think it would be fine to use the method that you have shown here which should protect against this in case it is ever able to happen in the future.